lib/audited/auditor.rb

```diff
@@ -185,16 +185,8 @@ def audited_changes
         end
       else
         def audited_changes
-          # either we are in a after save callback or not.
-          change_list = if saved_changes.present?
-            saved_changes.transform_values(&:first)
-          else
-            changed_attributes
-          end
-          change_list.except(*non_audited_columns).inject({}) do |changes, (attr, old_value)|
-            changes[attr] = [old_value, self[attr]]
-            changes
-          end
+          change_list = respond_to?(:changes_to_save) ? changes_to_save : changes
+          change_list.except(*non_audited_columns)
```

Comment:
It looks like there are other changes in the upstream version https://github.com/collectiveidea/audited/blob/v5.0.2/lib/audited/auditor.rb#L227-L239, are they not needed?

Reply:
I don't believe they are applicable, no. Our version is very far behind the upstream, and things like `redact_values` and `normalize_enum_changes` are new features that were introduced to the audited gem after we forked. As far as I can tell the `non_audited_columns` filtering is the only thing that we do in our fork of the gem that we've tried to keep feature-parity with.